### PR TITLE
✨ [amp-render] `placeholder` and `fallback` support

### PR DESCRIPTION
**Pull request by dmanek**
Partial for #32477

### CODE REVIEW

File: extensions/amp-render/1.0/test/test-amp-render.js [hunk 1]
```diff
@@ -517,5 +517,56 @@ describes.realWin(
...
+        </amp-render>
+      `;
+      doc.body.appendChild(element);
+
+      const text = await getRenderedData();
+      expect(text).to.equal('Loading data');
+      expect(fetchStub).not.to.be.called;
+    });
+
+    it.skip('should render a fallback', async () => {
```
Comment (caroqliu) [inline extensions/amp-render/1.0/test/test-amp-render.js:545]: Should this be skipped?

File: extensions/amp-render/1.0/test/test-amp-render.js [hunk 2]
```diff
@@ -517,5 +517,56 @@ describes.realWin(
...
+          <template type="amp-mustache"><p>Hello {{name}}</p></template>
+          <div placeholder>Loading data</div>
+          <div fallback>Failed</div>
+        </amp-render>
+      `;
+      doc.body.appendChild(element);
+
+      const text = await getRenderedData();
+      expect(text).to.equal('Failed');
+      // expect(fetchStub).not.to.be.called;
```
Comment (caroqliu) [inline extensions/amp-render/1.0/test/test-amp-render.js:569]: Is this supposed to be commented out?

File: src/custom-element.js [hunk 3]
```diff
@@ -1964,6 +1964,7 @@ function createBaseCustomElementClass(win, elementConnectedCallback) {
       const resourceState = this.getResource_().getState();
       // Do not show fallback before layout
       if (
+        !this.R1() &&
```
Comment (caroqliu) [inline src/custom-element.js:1967]: Why is this change necessary?

**Review by caroqliu — COMMENTED** [review]
Looks great, had a couple questions

File: src/custom-element.js (hunk 3 above)
Comment (dmanek) [reply src/custom-element.js:1967]: This is because R1 components don't use the resource system (there was a thread with @dvoytenko on this in our Bento eng)

File: extensions/amp-render/1.0/test/test-amp-render.js (hunk 2 above)
Comment (dmanek) [reply extensions/amp-render/1.0/test/test-amp-render.js]: Forgot to commit latest change.

File: src/custom-element.js (hunk 3 above)
Comment (caroqliu) [reply src/custom-element.js:1967]: Does this change require unit tests?

**Review by caroqliu — APPROVED** [review]

File: src/custom-element.js (hunk 3 above)
Comment (dmanek) [reply src/custom-element.js:1967]: Done.

Comment (samouri) [reply src/custom-element.js:1967]: I'm also confused why this is needed

**Review by samouri — APPROVED** [review]
Approving the change to `custom-element`

**Review by samouri — APPROVED** [review]
